src/atomate2/ase/schemas.py

```diff
@@ -0,0 +1,526 @@
+            steps=steps,
+            relax_kwargs=relax_kwargs,
+            optimizer_kwargs=optimizer_kwargs,
+        )
+
+        # Workaround for cases where the ASE optimizer does not correctly limit the
+        # number of steps for static calculations.
+        if steps <= 1:
+            steps = 1
+            n_steps = 1
+
+            if isinstance(input_mol_or_struct, Structure):
+                traj_method = "from_structures"
+            elif isinstance(input_mol_or_struct, Molecule):
+                traj_method = "from_molecules"
+
+            trajectory = getattr(PmgTrajectory, traj_method)(
+                [input_mol_or_struct],
+                frame_properties=[trajectory.frame_properties[0]],
+                constant_lattice=False,
+            )
+            output_mol_or_struct = input_mol_or_struct
+        else:
+            output_mol_or_struct = result.final_mol_or_struct
+
+        final_energy = trajectory.frame_properties[-1]["energy"]
+        final_energy_per_atom = final_energy / len(input_mol_or_struct)
+        final_forces = trajectory.frame_properties[-1]["forces"]
+        final_stress = trajectory.frame_properties[-1].get("stress")
+
+        ionic_steps = []
+        for idx in range(n_steps):
+            _ionic_step_data = {
+                key: trajectory.frame_properties[idx].get(key)
+                if key in ionic_step_data
+                else None
+                for key in ("energy", "forces", "stress")
+            }
+
+            current_mol_or_struct = (
+                trajectory[idx] if "mol_or_struct" in ionic_step_data else None
+            )
+
+            # include "magmoms" in :obj:`ionic_step` if the trajectory has "magmoms"
+            if "magmoms" in trajectory.frame_properties[idx]:
+                _ionic_step_data.update(
+                    {
+                        "magmoms": trajectory.frame_properties[idx]["magmoms"]
+                        if "magmoms" in ionic_step_data
+                        else None
+                    }
+                )
+
+            ionic_step = IonicStep(
+                mol_or_struct=current_mol_or_struct,
+                **_ionic_step_data,
+            )
+
+            ionic_steps.append(ionic_step)
+
+        objects: dict[AseObject, Any] = {}
+        if store_trajectory != StoreTrajectoryOption.NO:
+            # For VASP calculations, the PARTIAL trajectory option removes
+            # electronic step info. There is no equivalent for classical
+            # forcefields, so we just save the same info for FULL and
+            # PARTIAL options.
+            objects[AseObject.TRAJECTORY] = trajectory  # type: ignore[index]
+
+        output_doc = OutputDoc(
+            mol_or_struct=output_mol_or_struct,
+            energy=final_energy,
+            energy_per_atom=final_energy_per_atom,
+            forces=final_forces,
+            stress=final_stress,
+            ionic_steps=ionic_steps,
```

Comment:
`ionic_steps` stores the forces on every atom at every step and thus is roughly the size of the trajectory. In my tests with a moderately sized system (100 atoms, 400 timesteps) the TaskDocument size is exceeding the 16 MB limit and causing a memory error. I don't think this is coming up in testing because it's all being run with a `MemoryStore`.

The easiest way I can think to solve this would be by adding `"ionic_steps"` to the `_ASE_DATA_OBJECTS` and `_FORCEFIELD_DATA_OBJECTS` before they are passed to `ase_job` and `forcefield_job`. More generally, I think anything that scales with `n_steps` should probably be included in `data` so that long trajectories don't run into memory errors. It'd be nice if everything that scales with n_steps was included in a single object but that'd be a significant and probably unnecessary re-work.

In the future, (not needed in this PR) a way to only save a subset of trajectory steps would be desirable.

Reply:
Thanks @orionarcher , I'll take a look at your PR. For MD runs, only the energy should be saved in ionic steps by default

> In the future, (not needed in this PR) a way to only save a subset of trajectory steps would be desirable.

This is already possible with the `traj_interval` top-level kwarg to the ASE / forcefield MD makers, and by passing `relax_kwargs = {"interval" = <int> ,...}` through the `AseRelaxMaker`

Reply:
Ok awesome, apologies for missing that. Switching to sparser reporting has ameliorated some of the speed and database issues I was having.